From c62d20b054f4b8caead9b568fa4e642a263d1df5 Mon Sep 17 00:00:00 2001
From: thongsyho <thong.ho.px@rvc.renesas.com>
Date: Fri, 22 Jun 2018 11:18:46 +0700
Subject: [PATCH 217/442] ARM: dts: iwg20d-q7-dbcm-ca: Enable vin2 support

Enable vin2 support along with OV5640 CMOS camera on the
iWave RZ/G1{M,N} boards.

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/iwg20d-q7-dbcm-ca.dtsi | 41 ++++++++++++++++++++++++++++++++
 1 file changed, 41 insertions(+)

diff --git a/arch/arm/boot/dts/iwg20d-q7-dbcm-ca.dtsi b/arch/arm/boot/dts/iwg20d-q7-dbcm-ca.dtsi
index 7b293a8..1629737 100644
--- a/arch/arm/boot/dts/iwg20d-q7-dbcm-ca.dtsi
+++ b/arch/arm/boot/dts/iwg20d-q7-dbcm-ca.dtsi
@@ -66,6 +66,13 @@
 		output-high;
 		line-name = "tvp-power";
 	};
+
+	cam-power{
+		gpio-hog;
+		gpios = <3 GPIO_ACTIVE_HIGH>;
+		output-high;
+		line-name = "cam-power";
+	};
 };
 
 &hscif1 {
@@ -124,6 +131,19 @@
 			};
 		};
 	};
+
+	ov5640@3c {
+		compatible = "ovti,ov5640";
+		reg = <0x3c>;
+		remote = <&vin2>;
+
+		port {
+			ov5640: endpoint {
+				bus-width = <8>;
+				remote-endpoint = <&vin2ep>;
+			};
+		};
+	};
 };
 
 &pfc {
@@ -151,6 +171,11 @@
 		groups = "vin0_data8", "vin0_clk";
 		function = "vin0";
 	};
+
+	vin2_pins: vin2 {
+		groups = "vin2_data8", "vin2_clk";
+		function = "vin2";
+	};
 };
 
 &scif1 {
@@ -175,3 +200,19 @@
 		};
 	};
 };
+
+&vin2 {
+	status = "okay";
+	pinctrl-0 = <&vin2_pins>;
+	pinctrl-names = "default";
+
+	port {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		vin2ep: endpoint {
+			remote-endpoint = <&ov5640>;
+			bus-width = <8>;
+		};
+	};
+};
-- 
2.7.4

